tests/system/test_multigroups/test_multigroups.py

```diff
@@ -0,0 +1,259 @@
+                                   endpoint=f"/groups?groups_list={','.join(groups[1:])}")
+
+        # Remove any pre-existing multigroup (just in case they weren't removed after deleting the group).
+        mg_path = os.path.join(mg_folder_path, calculate_mg_name(groups))
+        for host in test_hosts:
+            if host_manager.run_shell(host, f"ls {mg_path}") == '':
+                break
+            host_manager.run_shell(host, cmd=f"rm -rf {mg_path}")
+
+
+# Fixtures
+
+@pytest.fixture(scope='function')
+def agent_healthcheck():
+    """Check if expected agents are active."""
+    failed_agents = set()
+    for _ in range(5):
+        for agent, _ in agent_groups.items():
+            result = host_manager.make_api_call(host=test_hosts[0], token=token, method='GET',
+                                                endpoint=f"/agents?status=active&name={agent}")
+            if result.get('json', {}).get('data', {}).get('total_affected_items', 0) != 1:
+                failed_agents.add(agent)
+                break
+            failed_agents.discard(agent)
+        else:
+            break
+        sleep(time_to_update)
+    else:
+        pytest.fail(f"These agents are not active: {', '.join(failed_agents)}")
+
+
+@pytest.fixture(scope='function')
+def clean_files():
+    """Remove test groups and multigroups before and after running a test."""
+    delete_groups()
+    yield
+    delete_groups()
+
+
+@pytest.fixture(scope='function')
+def create_multigroups():
+    """Create expected groups and multigroups."""
+    for agent_name, groups in agent_groups.items():
+        for idx, group in enumerate(groups):
+            if group != 'default':
+                # Create group.
+                response = host_manager.make_api_call(host=test_hosts[0], token=token, method='POST',
+                                                      endpoint='/groups', request_body={'group_id': group})
+                assert response['status'] == 200, f"Failed to create {group} group: {response}"
+
+                # Check that the multigroup folder does not exist yet.
+                sleep(time_to_update)
+                mg_path = os.path.join(mg_folder_path, calculate_mg_name(groups[:idx+1]))
+                assert '' == host_manager.run_shell(test_hosts[0], f"ls {mg_path}"), f"{mg_path} should not exist."
+
+                # Assign agent to group.
+                agent_id = get_agent_id(token=token, agent_name=agent_name)
+                response = host_manager.make_api_call(host=test_hosts[0], token=token, method='PUT',
+                                                      endpoint=f"/agents/{agent_id}/group/{group}")
+                assert response['status'] == 200, f"Failed to add {agent_name} ({agent_id}) to group: {response}"
+
+                # Check that the multigroup folder exists.
+                sleep(time_to_sync)
+                assert '' != host_manager.run_shell(test_hosts[0], f"ls {mg_path}"), f"{mg_path} should exist."
```

Comment:
```suggestion
                # Check that the multigroup folder does not exist yet.
                sleep(time_to_update)
                mg_name= calculate_mg_name(groups[:idx+1])
                assert mg_name not in host_manager.run_shell(test_hosts[0], f"ls {mg_folder_path}"), f"{mg_path} should not exist."

                # Assign agent to group.
                agent_id = get_agent_id(token=token, agent_name=agent_name)
                response = host_manager.make_api_call(host=test_hosts[0], token=token, method='PUT',
                                                      endpoint=f"/agents/{agent_id}/group/{group}")
                assert response['status'] == 200, f"Failed to add {agent_name} ({agent_id}) to group: {response}"

                # Check that the multigroup folder exists.
                sleep(time_to_sync)
                assert mg_name in host_manager.run_shell(test_hosts[0], f"ls {mg_folder_path}"), f"{mg_path} should exist."
```
We should check that the existent multigroup folder is the expected one.